modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -200,12 +199,12 @@ private void listenForTableChange() {
 
                 if (hasMetastorageLocally) {
                     var key = new Key(INTERNAL_PREFIX + tblId.toString());
-
                     futs.add(metaStorageMgr.invoke(
                         Conditions.key(key).value().eq(null),
                         Operations.put(key, tableView.name().getBytes(StandardCharsets.UTF_8)),
-                        Operations.noop()).thenCompose(res ->
-                        affMgr.calculateAssignments(tblId)));
+                        Operations.noop())
+                        .thenCompose(res -> schemaMgr.initNewSchemaForTable(tblId, tableView.name()))
```

Comment:
If you make it in the same manner as affinity, you would use a local event INITIALIZED/REMOVED (as affinity CALCULATED/REMOVED). Otherwise, you may be in circumstances where the metastorege has been consistently, but local structure (schema registry) has been not initialized.

Reply:
How are you suggest to synchronize 2 independent async events? e.g. Schema.REMOVED and Affinity.REMOVED.
Why you decide Affinity.REMOVED is terminal event for Table.drop() operation?

Is there any explanation or decision how components should communicate?

Reply:
For now I see the construction "affMgr.listen(AffinityEvent.REMOVED, ...)" which semantic is unclear and it has a bug.
If user will start/stop tables, affMgr.listeners collection will constantly grows, as noone removes useless listeners from it.